x-pack/plugin/spatial/src/main/java/org/apache/lucene/spatial3d/geom/GeoRegularConvexPolygon.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.apache.lucene.spatial3d.geom;
```

Comment:
We cannot have lucene packages in Elastic. We only did this to get around the fact that the class we extend is not public, but instead we should make a copy of the relevant non-public lucene code inside ES, and make an issue in Lucene to make that code public. Later once Lucene is available with the changes, we can remove the temporayr copy.